pkg/kubelet/apis/config/types.go

```diff
@@ -355,6 +355,10 @@ type KubeletConfiguration struct {
 	// The purpose of this format is make sure you have the opportunity to notice if the next release hides additional metrics,
 	// rather than being surprised when they are permanently removed in the release after that.
 	ShowHiddenMetricsForVersion string
+	// LogFormat Flag specifies the structure of log messages.
+	// default value of logFormat is `text`
+	// Refer [Logs Options](https://github.com/kubernetes/component-base/blob/master/logs/options.go) for more information.
+	LogFormat string
```

Comment:
don't we have several logging options (verbosity, vmodule, log file, log format, rotation, etc)? wouldn't it make sense to have a "logging" config struct that specified all of these together/consistently across components, rather than adding them piecemill? this seems similar to the LeaderElectionConfiguration / DebuggingConfiguration / ClientConnectionConfiguration component config options we have and use consistently

Reply:
You have listed klog options, we don't plan to support all klog options that you listed as we will replace klog at some point in future. 
Our current goal is to allow to switch implementation and decide in future which option should  be supported by all implementations.

I think it makes sense to put LogFormat into LogConfig object similar to https://github.com/kubernetes/component-base/blob/master/logs/options.go to allow for potential expansion.

@afrouzMashaykhi Could you add `LoggingConfig` struct in pkg/kubelet/apis/config/types.go and move `LogFormat` as a subfield to it?

Reply:
Ok, I will work on it.

Reply:
@liggitt I added "add loggingConfig struct to kubelet config" commit to add LoggingConfig struct which contains LoggingFormat for kubelet.